core/crypto/src/signer.rs

```diff
@@ -60,8 +60,8 @@ impl InMemorySigner {
         Self { account_id, public_key: secret_key.public_key(), secret_key }
     }
 
-    pub fn from_file(path: &Path) -> Self {
-        KeyFile::from_file(path).into()
+    pub fn from_file(path: &Path) -> std::io::Result<Self> {
+        Ok(KeyFile::from_file(path)?.into())
```

Comment:
Alternative, not necessary better, way to write this: 

```rust
KeyFile::from_file(path).map(InMemorySigner::from)
```

Reply:
yah I like that better too